### PR TITLE
Cleanup menu toolbars

### PR DESCRIPTION
**Pull request by tangele**
ref Yakindu/sctpro#1423

### CODE REVIEW

**Comment by tkutz** [comment]
Added "Do not merge yet" as this PR is not yet finished.

**Comment by tkutz** [comment]
No need for perspective listener, we can do everything in a declarative way (plugin.xml) => close

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramActionbarContributor.java [hunk 1]
```diff
@@ -26,15 +36,61 @@
 	@Override
 	public void init(IActionBars bars) {
 		super.init(bars);
+
 		// workaround for https://bugs.eclipse.org/bugs/show_bug.cgi?id=346648
 		bars.setGlobalActionHandler(GlobalActionId.SAVE, null);
 		bars.getToolBarManager().add(new DocumentationMenuAction());
-		
+
+		removeAppearanceButtonsInSimulationAndTestingPerspective(bars);
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramActionbarContributor.java]: Rename to something more readable, e.g. `showButtonsBasedOnPerspective` or even shorter `showButtonsByPerspective`

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramActionbarContributor.java [hunk 2]
```diff
@@ -26,15 +36,61 @@
...
+			}
+
+			@Override
+			public void perspectiveChanged(IWorkbenchPage page, IPerspectiveDescriptor perspective, String changeId) {
+				setToolbarButtons(bars, perspective);
+			}
+
+			private void setToolbarButtons(IActionBars bars, IPerspectiveDescriptor perspective) {
+				boolean isSimulationPers = ID_PERSPECTIVE_SCT_SIMULATION.equals(perspective.getId());
+				enableAppearanceButtons(bars.getToolBarManager(), !isSimulationPers);
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramActionbarContributor.java]: What about the SCT Testing perspective?
We need to clearify the requirements here, it is either

- show the buttons **only** in modeling perspective, or
- show the buttons **always but not in** simulation and testing perspective

currently you have implemented:

- show the buttons always but not in simulation perspective

Comment (tangele) [reply plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/StatechartDiagramActionbarContributor.java]: Is not in this Repo

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/DisableButtonsByPerspectiveListener.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	@Override
+	public void perspectiveChanged(IWorkbenchPage page, IPerspectiveDescriptor perspective, String changeId) {
+		setToolbarButtons(bars, perspective);
+	}
+
+	void setToolbarButtons(IActionBars bars, IPerspectiveDescriptor perspective) {
+		enableAppearanceButtons(bars.getToolBarManager(), !simulationPerspective(perspective));
+	}
+	
+	public boolean simulationPerspective(IPerspectiveDescriptor perspective) {
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/DisableButtonsByPerspectiveListener.java]: `doHideButtons(..)`

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/DisableButtonsByPerspectiveListener.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		setToolbarButtons(bars, perspective);
+	}
+
+	@Override
+	public void perspectiveChanged(IWorkbenchPage page, IPerspectiveDescriptor perspective, String changeId) {
+		setToolbarButtons(bars, perspective);
+	}
+
+	void setToolbarButtons(IActionBars bars, IPerspectiveDescriptor perspective) {
+		enableAppearanceButtons(bars.getToolBarManager(), !simulationPerspective(perspective));
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/DisableButtonsByPerspectiveListener.java]: `enableEditingButtons(..)`

File: plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/DisableButtonsByPerspectiveListener.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
+package org.yakindu.sct.ui.editor.editor;
```
Comment (tkutz) [inline plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/editor/DisableButtonsByPerspectiveListener.java]: add copyright header and author tag, then it should be fine

**Comment by tangele** [comment]
needs rebasing & conflict resolving

**Pull request merged**